tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -359,12 +361,23 @@ def __init__(self, policy):
 
     def validate(self):
         super(AzureEventGridMode, self).validate()
+        self._validate_is_arm_resource()
+        self._validate_event_matches_resource()
+
+    def _validate_is_arm_resource(self):
+        if not isinstance(self.policy.resource_manager, ArmResourceManager):
+            raise PolicyValidationError(
+                'The policy resource, {}, is not supported in event grid mode.'.format(
+                    self.policy.data['resource']))
+
+    def _validate_event_matches_resource(self):
         resource_type = self.policy.resource_manager.resource_type.resource_type
```

Comment:
this feels like it could be simplified, perhaps

`if self.policy.resource_type != GenericArmResource.type`

Reply:
The `resource_type` we need to check against events is different than the `resource_type` on the policy. The only time they are the same is for the GenericArmResource. So for this specific check, then sure it could be simplified. But to maintain the context of this validation method I prefer to keep it as is.